dlp/snippets/deid/deid_with_redact.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package deid
+
+// [START dlp_deidentify_redact]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyWithRedact de-identify the data by redacting with matched input values
+func deidentifyWithRedact(w io.Writer, projectID, inputStr string, infoTypeNames []string) error {
+	// projectID := "my-project-id"
+	// inputStr := "My name is Alicia Abernathy, and my email address is aabernathy@example.com."
+	// infoTypeNames := []string{"EMAIL_ADDRESS"}
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the content to be inspected.
+	var contentItem = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_Value{
+			Value: inputStr,
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
```

Comment:
praise: excellent use of comment to direct back to docs.

Reply:
Thank you 🙂